workspaces/homepage/plugins/dynamic-home-page/src/components/HomePage.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useMemo } from 'react';
+
+import { identityApiRef, useApi } from '@backstage/core-plugin-api';
+import { Content, EmptyState, Header, Page } from '@backstage/core-components';
+import { ClockConfig, HeaderWorldClock } from '@backstage/plugin-home';
+
+import useAsync from 'react-use/esm/useAsync';
+
+import { HomePageCardMountPoint } from '../types';
+import { ReadOnlyGrid } from './ReadOnlyGrid';
+import { LocalClock, LocalClockProps } from './LocalClock';
+
+export interface HomePageProps {
+  title?: string;
+  personalizedTitle?: string;
+  pageTitle?: string;
+  subtitle?: string;
+  localClock?: LocalClockProps;
+  worldClocks?: ClockConfig[];
+  cards?: HomePageCardMountPoint[];
+}
+
+// const getTimeBasedTitle = (): string => {
+//   const currentHour = new Date(Date.now()).getHours();
+//   if (currentHour < 12) {
+//     return 'Good morning {{firstName}}';
+//   } else if (currentHour < 17) {
+//     return 'Good afternoon {{firstName}}';
+//   }
+//   return 'Good evening {{firstName}}';
+// };
```

Comment:
I kept this because I hope that we will add this soon or at least in Dynamic Home Page plugin 1.2 ~ RHDH 1.6.